core/api/src/server.rs

```diff
@@ -215,7 +205,7 @@ pub trait ServerModulePlugin: Sized {
     type ConsensusItem: PluginConsensusItem;
     type VerificationCache: PluginVerificationCache;
 
-    fn init(&self);
+    fn init(&self, backend: &mut dyn InitHandle);
```

Comment:
Why supply a mutable reference instead of returning some init data that can then be processed? There's probably some type system reason for it, I just don't see it yet.

Reply:
I am actually confused by the rationale of the original version where:

```
pub handler:
        for<'a> fn(&'a M, serde_json::Value) -> BoxFuture<'a, Result<serde_json::Value, ApiError>>,
```

That `M` (being the module type) would have to be `&dyn IServerModule` maybe but then the handler would have to ... downcast it to concrete type to make it usable?  Then I thought it's kind of unnecessary to pass that argument as the module that was exposing these handler does have an access to `&self` already, so ... why bother passing at all? Maybe there's something about lifetimes there that I'm not seeing and will notice after I actually start porting the actual code.

But ultimately it can be done both ways, and it's just that `&mut dyn InitHandle` feels a bit natural to me, where all available options can be gathered and documented on a separate entity, instead of expanding the main module interface, and the module has freedom to use any logic it desires to register callbacks and call any other methods we might add to it.

Reply:
The handler type seems unrelated to my comment. What I meant is: what keeps us from returning a struct containing a list of handlers (and other things that `InitHandle` could set)? But I think I found a use case by now: checking if certain other modules are present (e.g. the LN module relies on the wallet module for block height consensus).